db/fixtures/ae_datastore/ManageIQ/System/CommonMethods/QuotaMethods.class/__methods__/requested.rb

```diff
@@ -27,68 +31,108 @@ def request_totals(template_totals, dialog_totals)
   else
-    options_value(prov_option, @miq_request, options_array)
+    options_value(args_hash)
   end
   options_array
 end
 
-def requested_number_of_cpus(number_of_vms, options_array)
-  cpu_in_request = get_option_value(@miq_request, :number_of_cpus)
-  if cpu_in_request.zero?
-    cpu_in_request = get_option_value(@miq_request, :number_of_sockets) *
-                     get_option_value(@miq_request, :cores_per_socket)
+def vm_provision_cloud?
+  @miq_request.source.respond_to?(:cloud) ? @miq_request.source.cloud : false
+end
+
+def flavor_obj(id)
+  vmdb_object('flavor', id)
+end
+
+def vm_prov_option_value(prov_option, options_array = [])
+  args_hash = {:prov_option   => prov_option,
+               :options_array => options_array,
+               :resource      => @miq_request,
+               :flavor        => flavor_obj(@miq_request.get_option(:instance_type)),
+               :number_of_vms => get_option_value(@miq_request, :number_of_vms),
+               :cloud         => vm_provision_cloud?}
+
+  case prov_option
+  when :vm_memory
+    args_hash[:prov_value] = args_hash[:number_of_vms] * get_option_value(@miq_request, :vm_memory)
+    request_hash_value(args_hash)
+  when :number_of_cpus
+    requested_number_of_cpus(args_hash)
+  when :storage
+    requested_storage(args_hash)
+  else
+    options_value(args_hash)
   end
-  total_cpu = number_of_vms * cpu_in_request
-  set_requested_value(prov_option, total_cpu,
-                      @miq_request.get_option(:instance_type), options_array)
+  options_array
+end
+
+def requested_number_of_cpus(args_hash)
+  cpu_in_request = get_option_value(args_hash[:resource], :number_of_sockets) *
+                   get_option_value(args_hash[:resource], :cores_per_socket)
+  cpu_in_request = get_option_value(args_hash[:resource], args_hash[:number_of_cpus]) if cpu_in_request.zero?
+  args_hash[:prov_value] = args_hash[:number_of_vms] * cpu_in_request
+  request_hash_value(args_hash)
 end
 
 def bytes_to_megabytes(bytes)
   bytes / 1024**2
 end
 
-def requested_storage(prov_option, number_of_vms, options_array)
-  vm_size = bytes_to_megabytes(@miq_request.vm_template.provisioned_storage)
-  total_storage = number_of_vms * vm_size
-  set_requested_value(prov_option, total_storage,
-                      @miq_request.get_option(:instance_type), options_array)
+def vmdb_object(model, id)
+  $evm.vmdb(model, id.to_i) if model && id
+end
+
+def requested_storage(args_hash)
+  vm_size = bytes_to_megabytes(args_hash[:resource].vm_template.provisioned_storage)
+  args_hash[:prov_value] = args_hash[:number_of_vms] * vm_size
+  request_hash_value(args_hash)
 end
 
-def options_value(prov_option, resource, options_array)
-  return unless resource.respond_to?('get_option')
-  set_requested_value(prov_option, resource.get_option(prov_option),
-                      resource.get_option(:instance_type), options_array)
+def request_object?(object)
+  return false if object.nil?
```

Comment:
@tinaafitz This check is not needed since `object.respond_to?('get_option')` would already return `false` if object is nil.